packages/elements/src/components/file/file.component.ts

```diff
@@ -211,18 +212,48 @@ export class FileComponent extends LitElement {
           }
         }
       });
+
+      if (this.selectedMutant !== undefined) {
+        this.dispatchEventIfMutantIsUpdated();
+      }
     }
     if ((changes.has('model') && this.model) || changes.has('selectedMutantStates')) {
       this.mutants = this.model.mutants
         .filter((mutant) => this.selectedMutantStates.includes(mutant.status))
         .sort((m1, m2) => (gte(m1.location.start, m2.location.start) ? 1 : -1));
-      if (this.selectedMutant && !this.mutants.includes(this.selectedMutant)) {
+
+      if (
+        this.selectedMutant &&
+        !this.mutants.includes(this.selectedMutant) &&
+        changes.has('selectedMutantStates') &&
+        // This extra check is to allow mutants that have been opened before, to stay open when a realtime update comes through
+        this.selectedMutantsHaveChanged(changes.get('selectedMutantStates'))
+      ) {
         this.toggleMutant(this.selectedMutant);
       }
     }
     super.update(changes);
   }
 
+  private dispatchEventIfMutantIsUpdated(): void {
+    const theSameMutant = this.model.mutants.find((mutant) => mutant.id === this.selectedMutant?.id);
+    // TODO: deep eq?
+    if (theSameMutant === undefined || this.selectedMutant?.status === theSameMutant.status) {
```

Comment:
Should we use a package for this? Or should I manually do this check (since it is only for the mutant)?